lib/utils.pm

```diff
@@ -1663,6 +1664,55 @@ sub script_output_retry {
     die("Waiting for Godot: $cmd") if $die;
 }
 
+
+=head2 validate_script_output_retry
+
+ validate_script_output_retry($cmd, $check, [retry => $retry], [delay => $delay], [timeout => $timeout]);
+
+Repeat command until validate_script_output succeeds or die. Return the output of the command on success.
+
+C<$retry> refers to the number of retries and defaults to C<10>.
+
+C<$delay> is the time between retries and defaults to C<30>.
+
+If the command doesn't succeed after C<$retry> retries,
+this function will die.
+
+Example:
+
+ validate_script_output_retry('ping -c1 -W1 machine', m/1 packets transmitted/, retry => 5, delay => 60);
+
+=cut
+sub validate_script_output_retry {
+    my ($cmd, $check, %args) = @_;
+    $args{retry} //= 10;
+    $args{delay} //= 30;
+    $args{timeout} //= 90;
+    $args{proceed_on_failure} //= 1;
+    my $retry = delete $args{retry};
+    my $delay = delete $args{delay};
+    my $ret;
+
+    my $exec = "timeout $args{timeout} $cmd";
+    # Exclamation mark needs to be moved before the timeout command, if present
+    if (substr($cmd, 0, 1) eq "!") {
+        $cmd = substr($cmd, 1);
+        $cmd =~ s/^\s+//;    # left trim spaces after the exclamation mark
+        $exec = "! timeout $args{timeout} $cmd";
+    }
+
+    $args{timeout} += 3;    # timeout for script_run must be larger than for the 'timeout ...' command
+    my $timeout = delete $args{timeout};
```

Comment:
As you pointend me to 
the output looks like:
```
# cat > /tmp/scriptIzNvi.sh << 'EOT_IzNvi'; echo IzNvi-$?-
> timeout 300 podman run --entrypoint /usr/lib/zypp/plugins/services/container-suseconnect-zypp -i registry.suse.com/suse/sles12sp3 lp
> EOT_IzNvi
IzNvi-0-
# echo IzNvi; bash -oe pipefail /tmp/scriptIzNvi.sh ; echo SCRIPT_FINISHEDIzNvi-$?-
IzNvi
cat > /tmp/scriptIzNvi.sh << 'EOT_IzNvi'; echo IzNvi-$?-
timeout 300 podman run --entrypoint /usr/lib/zypp/plugins/services/container-suseconnect-zypp -i registry.suse.com/suse/sles12sp3 lp
EOT_IzNvi
echo IzNvi; bash -oe pipefail /tmp/scriptIzNvi.sh ; echo SCRIPT_FINISHEDIzNvi-$?-
cat > /tmp/scriptIzNvi.sh << 'EOT_IzNvi'; echo IzNvi-$?-
timeout 300 podman run --entrypoint /usr/lib/zypp/plugins/services/container-suseconnect-zypp -i registry.suse.com/suse/sles12sp3 lp
EOT_IzNvi
echo IzNvi; bash -oe pipefail /tmp/scriptIzNvi.sh ; echo SCRIPT_FINISHEDIzNvi-$?-
cat > /tmp/scriptIzNvi.sh << 'EOT_IzNvi'; echo IzNvi-$?-
timeout 300 podman run --entrypoint /usr/lib/zypp/plugins/services/container-suseconnect-zypp -i registry.suse.com/suse/sles12sp3 lp
EOT_IzNvi
echo IzNvi; bash -oe pipefail /tmp/scriptIzNvi.sh ; echo SCRIPT_FINISHEDIzNvi-$?-
cat > /tmp/scriptIzNvi.sh << 'EOT_IzNvi'; echo IzNvi-$?-
timeout 300 podman run --entrypoint /usr/lib/zypp/plugins/services/container-suseconnect-zypp -i registry.suse.com/suse/sles12sp3 lp
EOT_IzNvi
echo IzNvi; bash -oe pipefail /tmp/scriptIzNvi.sh ; echo SCRIPT_FINISHEDIzNvi-$?-
```
Which for me looks like podman somehow do not stop on `TERM` signal.

My idea is to add `--kill-after=DURATION` to your command, like:
```suggestion
    my $delay = delete $args{delay};
    my $timeout = delete $args{timeout};
    my $ret;

    my $exec = "timeout --kill-after 5s ${timeout}s";
    # Exclamation mark needs to be moved before the timeout command, if present
    if (substr($cmd, 0, 1) eq "!") {
        $cmd = substr($cmd, 1);
        $cmd =~ s/^\s+//;    # left trim spaces after the exclamation mark
        $exec = "! $exec";
    }
    $exec = "$exec $cmd"

    $timeout += 8;    # timeout for script_run must be larger than for the 'timeout ...' command
```